BSP/STM32F413/Src/BSP_PWM.c

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
+
+#include "BSP_PWM.h"
+
+GPIO_InitTypeDef GPIO_INIT_STRUCT; //struct used to initialize pins
+TIM_OCInitTypeDef TIMER_OC_STRUCT; //struct used to configure output compare for timers
+TIM_TimeBaseInitTypeDef TIMER_INIT_STRUCT; //struct used to initialize PWM timers
+
+void BSP_PWM_Init(void){
+    //we will initialize all pins pertaining to fans and contactors for PWM here
+
+    //Enable TIM 3,8,12
+    RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM3, ENABLE);
+    RCC_APB2PeriphClockCmd(RCC_APB2Periph_TIM8, ENABLE);
+    RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM12, ENABLE);
+
+    //Enable Port Clocks
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOA, ENABLE);
```

Comment:
From the github UI, the tabs seem wonky. Not sure if they actually are but make sure everything is aligned properly (in terms of whitespace).

Reply:
idk why they wonk on github, on vscode it's fine